server/src/main/java/org/elasticsearch/index/mapper/BlockSourceReader.java

```diff
@@ -22,13 +22,23 @@
 import java.io.IOException;
 import java.util.ArrayList;
 import java.util.List;
+import java.util.Set;
 
 /**
  * Loads values from {@code _source}. This whole process is very slow and cast-tastic,
  * so it doesn't really try to avoid megamorphic invocations. It's just going to be
  * slow.
  */
 public abstract class BlockSourceReader implements BlockLoader.RowStrideReader {
+
+    // _ignored_source is needed ofr synthetic source is needed for, in case stored source (default) is used,
+    // then it just doesn't get loaded.
+    static final StoredFieldsSpec NEEDS_SOURCE_AND_IGNORED_SOURCE = new StoredFieldsSpec(
```

Comment:
Ideally, we should avoid requesting _source and only read `_ignored_source` when synthetic source is enabled. However, we should get this in to ensure correctness, and make the blockloader more selective in a follow-up. I can help with that.

Reply:
A few unit compute engine tests failed, and instead of adjusting the tests, I made the block loaders a little more selective, based on whether synthetic source is used. See: 6702a2e7634240a5db878bf00284106d53f72288